common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -34,6 +34,27 @@
  * Utility class for collation-aware UTF8String operations.
  */
 public class CollationAwareUTF8String {
+
+  public static boolean lowercaseMatchFrom(final UTF8String l, final UTF8String r, int pos) {
+    if (pos < 0) return false;
```

Comment:
Why do we return false in this case?
Also why not make this an assert and make sure we never call this method with a negative `pos`?

Reply:
`lowercaseMatchFrom` here is part of a public interface, so I think we do need to establish some rules on how it will behave when given a negative position

the ways I see of doing this are:

- return false by default
- throw an exception
- assert that pos is not negative


In this case, I agree that assert is the best choice, so I'll implement that